server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java

```diff
@@ -129,94 +128,72 @@ public void onFailure(Exception e) {}
-
-        ensureNoPendingMasterTasks().actionGet(TimeValue.timeValueSeconds(30));
-        ActionFuture<Res> future = req.execute();
-
-        // cancel the first cluster state update produced by the request above
-        assertBusy(() -> assertTrue(masterCoordinator.cancelCommittedPublication()));
-        // await and cancel any other forked cluster state updates that might be produced by the request
-        var task = ensureNoPendingMasterTasks();
-        while (task.isDone() == false) {
-            masterCoordinator.cancelCommittedPublication();
-            Thread.onSpinWait();
-        }
-        task.actionGet(TimeValue.timeValueSeconds(30));
-
-        return future;
-    }
-
-    private PlainActionFuture<Void> ensureNoPendingMasterTasks() {
-        var future = new PlainActionFuture<Void>();
-        internalCluster().getCurrentMasterNodeInstance(ClusterService.class)
-            .submitUnbatchedStateUpdateTask(
-                "ensureNoPendingMasterTasks",
-                new ClusterStateUpdateTask(Priority.LANGUID, TimeValue.timeValueSeconds(30)) {
-
-                    @Override
-                    public ClusterState execute(ClusterState currentState) {
-                        return currentState;
-                    }
-
-                    @Override
-                    public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
-                        future.onResponse(null);
-                    }
-
-                    @Override
-                    public void onFailure(Exception e) {
-                        future.onFailure(e);
-                    }
-                }
-            );
-        return future;
-    }
-
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/102947")
     public void testDeleteCreateInOneBulk() throws Exception {
         internalCluster().startMasterOnlyNode();
         String dataNode = internalCluster().startDataOnlyNode();
         assertFalse(clusterAdmin().prepareHealth().setWaitForNodes("2").get().isTimedOut());
         prepareCreate("test").setSettings(Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)).get();
         ensureGreen("test");
 
-        // block none master node.
-        BlockClusterStateProcessing disruption = new BlockClusterStateProcessing(dataNode, random());
-        internalCluster().setDisruptionScheme(disruption);
+        final var masterClusterService = internalCluster().clusterService(internalCluster().getMasterName());
+        final var dataNodeClusterService = internalCluster().clusterService(dataNode);
+
+        final var originalIndexUuid = masterClusterService.state().metadata().index("test").getIndexUUID();
+        final var uuidChangedListener = ClusterServiceUtils.addTemporaryStateListener(dataNodeClusterService, cs -> {
+            // NB throws a NPE which fails the test if the data node sees the intermediate state with the index deleted
```

Comment:
I would suggest to use an explicit null check rather than exception and extra change in ClusterServiceUtils. Especially if such NPE could be logged cluttering the test output

Reply:
Not sure I follow. The exception will be logged indeed but it's not clutter, it's a meaningful test failure.

Reply:
Ok, I missed that this is a meaningful failure and such state is not expected. Disregard my comment